pkg/reconciler/openshift/tektonaddon/pipelinesascode.go

```diff
@@ -86,3 +93,26 @@ func (r *Reconciler) ensurePAC(ctx context.Context, ta *v1alpha1.TektonAddon) er
 
 	return nil
 }
+
+// pacTriggerTemplateStepImages returns a map[string]string with key as step name and
+// value as image name to be replaced with, from the env vars that start with
+// IMAGE_PAC_
+func pacTriggerTemplateStepImages() map[string]string {
+	triggerTemplateSteps := make(map[string]string)
+
+	// pacImage is a map[string]string which will have key-values like
+	// "TRIGGERTEMPALTE_APPLY_AND_LAUNCH": "registry.example.io/pac-image"
+	pacImages := common.ToLowerCaseKeys(common.ImagesFromEnv(common.PacImagePrefix))
+	for env, image := range pacImages {
+		prefix := "triggertemplate_"
+		if strings.HasPrefix(env, prefix) {
+			// step 3: "apply-and-launch": "registry.example.io/pac-image"
+			triggerTemplateSteps[
+			// step 2: apply_and_launch --> apply-and-launch
+			strings.ReplaceAll(
```

Comment:
is not better to have this replace on step name inside transformer, so any one using this function has not need to do this every time and also it will be consistent with other transformers

Reply:
which replace are you referring to that happens in other transformers?
https://github.com/concaf/operator/blob/307eafa4a3cce8f01e2354e651c50afb84f3ae93/pkg/reconciler/common/transformers.go#L262
^ `formKey` converts container names to env var names so we can match it, but that's not what I'm doing here. I'm normalizing the step names before so the transformer gets the right step names instead of having to modify it later.

anyway, I'd rather have the transformer get the correct normalized step names as input instead of having the modify and match inside it.

Reply:
Yes the idea behind this is, anyone calling this function does not need to do a modification for all calls, rather the transformer is doing the normalization and this is the same thing for all transformers

Reply:
normalization is happening out of the transformer in other calls except the `formKey` 🤔

Reply:
I don't have any other strong points behind keeping it inside and outside apart from these two

1. Consistency with the codebase
2. Benefit of having every caller does not need to do the modification

I will let the owners of codebase take the call cc @tektoncd/operator-maintainers

Reply:
we can discuss this in WG and can do a follow-up PR for the pattern we agree

Reply:
I think the other transformers are a bit inconsistent in the way of normalizing input names. Let's take an example of `common.DeploymentImages(images)` https://github.com/tektoncd/operator/blob/307eafa4a3cce8f01e2354e651c50afb84f3ae93/pkg/reconciler/common/transformers.go#L177

It takes `images` as input which I'd expect to be a mapping of `container name: image name` and this is true to an extent but not fully.

Some normalization of the container names happens outside the transformer here:
https://github.com/tektoncd/operator/blob/307eafa4a3cce8f01e2354e651c50afb84f3ae93/pkg/reconciler/kubernetes/tektonpipeline/tektonpipeline.go#L338 where the prefix is removed, lowercased, etc but `_` is not yet replaced by `-` which happens inside the transformer instead - https://github.com/tektoncd/operator/blob/307eafa4a3cce8f01e2354e651c50afb84f3ae93/pkg/reconciler/common/transformers.go#L230

I didn't want this inconsistent normalization to be happening for this transformer and I think it's the caller's responsibility to provide normalized container names and not the transformer's; unless you want to tie the transformer to env variables which I don't think is ideal because it reduces the transformers' scope by quite a bit and it cannot be called for non env var cases.

At some point, we should consider doing normalization outside of other transformers as well 🤔